### DISCUSSION TITLE
如何配置站点图标 favicon?

### DISCUSSION THREAD

**Discussion by zzx58-027** [umijs/umi#10846] [Q&A]
版本 Umi4
<img width="757" alt="image" src="https://user-images.githubusercontent.com/77034104/227475613-c3fc8716-80a9-4962-b0d7-770acf77f25d.png">
根据文档，在 src/ 目录下提供了 favicon.png/ico 两个文件，生成产物中也能够正常看到这两个文件有链接上，但是使用 http-server 运行查看，图标就是不生效，是什么原因？
<img width="176" alt="image" src="https://user-images.githubusercontent.com/77034104/227476168-a6162896-0407-49c7-a32d-ded5b3610f87.png">
<img width="1155" alt="image" src="https://user-images.githubusercontent.com/77034104/227476265-80d6ac7d-9566-4cd9-b22e-c60f31d198b8.png">
devtools 中查看，umi用的页面和 产物的 index.html 不一样？？

**Comment by zzx58-027** [comment]
另一个问题是在 .umirc.ts 文件中配置 links 选项，配置 manifest.json,但是 build 过程后 manifest.json 文件并没有被包含在产物当中，怎么做才能让产物中包含有 manifest.json 文件？
想要打包进产物的文件通过什么地方进行配置？比如根目录下我还存放了 logo.png 文件。

**Reply by fz6m** [reply]
1. favicon 估计是浏览器缓存，alt / cmd + shift + f5 刷缓存看。

2. 配了 links 只是生成这些 header ，但是文件还要你自己拷贝，放到 `public` 文件夹会自己拷贝到产物里，或者用 [`copy`](https://umijs.org/docs/api/config#copy) 选项定向拷贝。

**Reply by zzx58-027** [reply]
感谢。附加一个问题，这个文件提示找不到，我不知道这个问题是关联在什么配置项上的。
<img width="692" alt="image" src="https://user-images.githubusercontent.com/77034104/227827363-66ab8df9-f23a-427b-8015-008f4e9810ec.png">

**Reply by fz6m** [reply]
你用到了外部的 `drawer.css` 这个全局静态资源，但是他没有 css source map ，这是正常的，仅在开发时有提示（因为开发时需要依靠 source map 来映射回源代码位置）。

**Reply by zzx58-027** [reply]
再次感谢如此及时的回复！
